docs/guides/environmental-variables.md

```diff
@@ -0,0 +1,57 @@
+---
+title: CLI
+---
+
+Environment variables are key-value pairs that are stored in our operating systems. Many of them come by default, and we can also create our own. They are used to store information that is used by our operating system or by other programs. For example, the `PATH` variable stores a list of directories that the operating system should look in when we run a command in the terminal. The `HOME` variable stores the path to our home directory.
+
+The same can be used for building softwares aswell. Configuration values are often stored in environmental variables.
+
+## Need of Environmental Variables
+
+Applications use multiple external tools, authentication methods, and numerous configurations. Hence, for proper functioning, we need our code to use these values properly.
+
+Consider a simple scenario of [JWT authentication](https://jwt.io/). Typically, when signing the tokens for our users, we will need the following configuration set:
+
+- **Expiry time**: The time after which the token will expire
+- **Secret key**: The secret key that will be used to encrypt the token
+- **Issuer**: The name of the issuer of the token. This is generally the organization's name.
+
+To go about this, there are broadly two ways:
+
+1. **Hardcode the values in our code** \
+   This is perhaps the easiest approach to solve the problem, but the most dangerous (and inefficient) one. Once you hardcode the values in your codebase, everyone who works on the code will be able to see your sensitive information, which poses a massive security issue. Also, if you want to change the values, you will have to change the code and redeploy the application, which is not ideal.
+
+2. **Store the values in environmental variables** \
+   This is the best approach to solve the problem. You can store the sensitive values in the OS of the server running your application, and during the runtime of your application, you can access these values from the OS. All programming languages have excellent support for this. This way, you can keep your sensitive information safe, and also change the values without having to change anything in your code.
+
+## Environmental variables in Tailcall
+
+With Tailcall, you can seamlessly fit in environmental variables into your GraphQL schema. Tailcall supports this by providing a `env` [Context](context.md) variable. This Context is then shared across all the operators, which you will use to resolve values in your schema.
+
+For example, you can do the following:
+
+```graphql showLineNumbers
+type Query {
+  users: [User]! @http(baseUrl: "{{env.API_ENDPOINT}}", path: "/users")
```

Comment:
- Expand on how this is better than hard coding the baseURL. For eg: You could have different URLs in Staging and Production,  etc.

- Can add this info - Env Variable can be used in all the places wherever we use mustache templates (in better wording).

Reply:
sure will do